mmcv/image/io.py

```diff
@@ -239,20 +258,52 @@ def imfrombytes(content, flag='color', channel_order='bgr', backend=None):
         return img
 
 
-def imwrite(img, file_path, params=None, auto_mkdir=True):
+def imwrite(img,
+            file_path,
+            params=None,
+            auto_mkdir=True,
+            file_client_args=None):
     """Write image to file.
 
+    Note:
+        In v1.4.1 and later, add `file_client_args` parameters.
+
     Args:
         img (ndarray): Image array to be written.
         file_path (str): Image file path.
         params (None or list): Same as opencv :func:`imwrite` interface.
         auto_mkdir (bool): If the parent folder of `file_path` does not exist,
             whether to create it automatically.
+        file_client_args (dict | None): Arguments to instantiate a
+            FileClient. See :class:`mmcv.fileio.FileClient` for details.
+            Default: None.
 
     Returns:
         bool: Successful or not.
+
+    Examples:
+        >>> ret = mmcv.imwrite(img, '/path/to/img.jpg')
+        >>> ret = mmcv.imwrite(img, 's3://bucket/img.jpg')
+        >>> ret = mmcv.imwrite(img, 's3://bucket/img.jpg', file_client_args={
+        ...     'backend': 'petrel'})
     """
-    if auto_mkdir:
-        dir_name = osp.abspath(osp.dirname(file_path))
-        mkdir_or_exist(dir_name)
-    return cv2.imwrite(file_path, img, params)
+    assert is_str(file_path)
+    file_client = FileClient.infer_client(file_client_args, file_path)
+    if not hasattr(file_client.client, 'put'):
+        raise AttributeError(
+            f"{file_client.name} doesn't contain the `put` method")
+    try:
+        img_ext = osp.splitext(file_path)[-1]
+        _, img_buff = cv2.imencode(img_ext, img, params)
+        # The HardDiskBackend automatically mkdir by default,it needs to be
+        # judged individually
+        if not auto_mkdir and file_client.name == 'HardDiskBackend':
```

Comment:
Note that `file_client.put` should follow a Protocol that makes directory automatically if needed.